packages/editor-ui/src/components/FixedCollectionParameter.vue

```diff
@@ -1,220 +1,201 @@
-			mutableValues: {} as Record<string, INodeParameters[]>,
-		};
-	},
-	computed: {
-		getPlaceholderText(): string {
-			const placeholder = this.$locale.nodeText().placeholder(this.parameter, this.path);
-			return placeholder ? placeholder : this.$locale.baseText('fixedCollectionParameter.choose');
-		},
-		getProperties(): INodePropertyCollection[] {
-			const returnProperties = [];
-			let tempProperties;
-			for (const name of this.propertyNames) {
-				tempProperties = this.getOptionProperties(name);
-				if (tempProperties !== undefined) {
-					returnProperties.push(tempProperties);
-				}
-			}
-			return returnProperties;
-		},
-		multipleValues(): boolean {
-			return !!this.parameter.typeOptions?.multipleValues;
-		},
-		parameterOptions(): INodePropertyCollection[] {
-			if (this.multipleValues && isINodePropertyCollectionList(this.parameter.options)) {
-				return this.parameter.options;
-			}
+import { computed, ref, watch, onBeforeMount } from 'vue';
+import { useI18n } from '@/composables/useI18n';
 
-			return (this.parameter.options as INodePropertyCollection[]).filter((option) => {
-				return !this.propertyNames.includes(option.name);
-			});
-		},
-		propertyNames(): string[] {
-			return Object.keys(this.mutableValues || {});
-		},
-		sortable(): boolean {
-			return !!this.parameter.typeOptions?.sortable;
-		},
-	},
-	watch: {
-		values: {
-			handler(newValues: Record<string, INodeParameters[]>) {
-				this.mutableValues = deepCopy(newValues);
-			},
-			deep: true,
-		},
-	},
-	created() {
-		this.mutableValues = deepCopy(this.values);
+const locale = useI18n();
+
+const props = defineProps<{
+	nodeValues: INodeParameters;
+	parameter: INodeProperties;
+	path: string;
+	values: Record<string, INodeParameters[]>;
+	isReadOnly: boolean | undefined;
+}>();
+
+const emit = defineEmits<{
+	valueChanged: [value?: any];
+}>();
+
+const getPlaceholderText = computed(() => {
+	const placeholder = locale.nodeText().placeholder(props.parameter, props.path);
+	return placeholder ? placeholder : locale.baseText('fixedCollectionParameter.choose');
+});
+const mutableValues = ref({} as Record<string, INodeParameters[]>);
+const selectedOption = ref<string | null | undefined>(null);
+const propertyNames = computed(() => {
+	return Object.keys(mutableValues.value || {});
+});
+const getProperties = computed(() => {
+	const returnProperties = [];
+	let tempProperties;
+	for (const name of propertyNames.value) {
+		tempProperties = getOptionProperties(name);
+		if (tempProperties !== undefined) {
+			returnProperties.push(tempProperties);
+		}
+	}
+	return returnProperties;
+});
+const multipleValues = computed(() => {
+	return !!props.parameter.typeOptions?.multipleValues;
+});
+const parameterOptions = computed(() => {
+	if (multipleValues && isINodePropertyCollectionList(props.parameter.options)) {
+		return props.parameter.options;
+	}
+
+	return (props.parameter.options as INodePropertyCollection[]).filter((option) => {
```

Comment:
I think this type cast is avoidable by doing `props.parameter.options ?? []`, would be safer too 😄